conn_test.go

```diff
@@ -1740,49 +1736,6 @@ func TestProtocolVersionValidation(t *testing.T) {
 							},
 						},
 					},
-					{
-						Header: recordlayer.Header{
-							Version:        protocol.Version1_2,
-							SequenceNumber: 2,
-						},
-						Content: &handshake.Handshake{
-							Header: handshake.Header{
-								MessageSequence: 2,
-							},
-							Message: &handshake.MessageCertificate{},
-						},
-					},
-					{
-						Header: recordlayer.Header{
-							Version:        protocol.Version1_2,
-							SequenceNumber: 3,
-						},
-						Content: &handshake.Handshake{
-							Header: handshake.Header{
-								MessageSequence: 3,
-							},
-							Message: &handshake.MessageServerKeyExchange{
-								EllipticCurveType:  elliptic.CurveTypeNamedCurve,
-								NamedCurve:         elliptic.X25519,
-								PublicKey:          localKeypair.PublicKey,
-								HashAlgorithm:      hash.SHA256,
-								SignatureAlgorithm: signature.ECDSA,
-								Signature:          make([]byte, 64),
-							},
-						},
-					},
-					{
-						Header: recordlayer.Header{
-							Version:        protocol.Version1_2,
-							SequenceNumber: 4,
-						},
-						Content: &handshake.Handshake{
-							Header: handshake.Header{
-								MessageSequence: 4,
-							},
-							Message: &handshake.MessageServerHelloDone{},
-						},
-					},
```

Comment:
Is it related to session resumption? If not, please separate it to another PR.

Reply:
Actually , it do be related.

In the flight3handler, we need to firstly check the ServerHello message. And then check the following messages according whether  do the resumption.

If the ServerHello message version is invalid, the flght3handler.flight3Parse will return with errUnsupportedProtocolVersion and there is no opportunity to read the following message. And there is no need to read them as well.

https://github.com/pion/dtls/pull/369/files#diff-393727b0114931101cdcbeaf54958a69e7bede28a147364cd963c4eb4bf3e782R44-R46

Reply:
It would be better to check RFC to confirm which behavior is correct.

Reply:
hi @at-wat , according to [rfc5246#section-7.2.2](https://datatracker.ietf.org/doc/html/rfc5246#section-7.2.2)

> Upon transmission or receipt of a fatal alert message, both
   parties immediately close the connection.  Servers and clients MUST
   forget any session-identifiers, keys, and secrets associated with a
   failed connection.  Thus, any connection terminated with a fatal
   alert MUST NOT be resumed.

When the client receive a ServerHello message with invalid version, it should send the protocol_version alert message, which is a fatal alert. So the client should close the connection immediately. And the server's message behind ServerHello may be not processed.